yarl/_url.py

```diff
@@ -1025,6 +1026,16 @@ def update_query(self, *args, **kwargs):
             self._val._replace(query=self._get_str_query(query) or ""), encoded=True
         )
 
+    def without_query_params(self, *query_params: str) -> "URL":
+        """Remove some keys from query part and return new URL."""
+        params_to_remove = set(query_params) & set(self.query.keys())
+        if not params_to_remove:
+            return URL(self._val._replace(query=self.query_string), encoded=True)
```

Comment:
Why make not just `URL(self)`?

Reply:
Thanks.

Reply:
Tried as you suggested, but `URL(self)` does not create new object, due to [this logic](https://github.com/aio-libs/yarl/blob/0698dc947772786fcff467b12cfb1f3921c7ff31/yarl/_url.py#L169C1-L170)

Reply:
Why does it need to be a new object? Since `URL` is immutable we should be fine with returning `self` if there are no changes